src/lib/dlt_user.c

```diff
@@ -3736,10 +3736,12 @@ static void dlt_user_cleanup_handler(void *arg)
     DLT_SEM_FREE();
 }
 
-void dlt_user_housekeeperthread_function(__attribute__((unused)) void *ptr)
+void dlt_user_housekeeperthread_function(void *ptr)
 {
     struct timespec ts;
     bool in_loop = true;
+    bool *thread_running = (bool*)ptr;
+    *thread_running = true;
```

Comment:
Hello @alexmohr ,
in which platform have you faced this issue?
I guess the assignment to true needs to be shifted below pthread_sigmask (for Android) and calling pthread_cleanup_push(). Otherwise a race condition still exists.

Reply:
Issue occurs on linux but only in unit tests when running them on a very fast machine. I moved the call and replaced the sleep with a condition variable

Reply:
Looks good to me. Thank you very much.